global/scripts/TopupPreprocessingAll.sh

```diff
@@ -136,16 +185,71 @@ if [ ! $GradientDistortionCoeffs = "NONE" ] ; then
       --out=${WD}/PhaseTwo_gdc \
       --owarp=${WD}/PhaseTwo_gdc_warp
 
+  if [ $Phase2ndDir = TRUE ] ; then
+    ${GlobalScripts}/GradientDistortionUnwarp.sh \
+      --workingdir=${WD} \
+      --coeffs=${GradientDistortionCoeffs} \
+      --in=${WD}/PhaseOne2 \
+      --out=${WD}/PhaseOne2_gdc \
+      --owarp=${WD}/PhaseOne2_gdc_warp
+
+    ${GlobalScripts}/GradientDistortionUnwarp.sh \
+      --workingdir=${WD} \
+      --coeffs=${GradientDistortionCoeffs} \
+      --in=${WD}/PhaseTwo2 \
+      --out=${WD}/PhaseTwo2_gdc \
+      --owarp=${WD}/PhaseTwo2_gdc_warp
+  fi
+
   if ((UseJacobian))
   then
     ${FSLDIR}/bin/fslmaths ${WD}/PhaseOne_gdc -mul ${WD}/PhaseOne_gdc_warp_jacobian ${WD}/PhaseOne_gdc
     ${FSLDIR}/bin/fslmaths ${WD}/PhaseTwo_gdc -mul ${WD}/PhaseTwo_gdc_warp_jacobian ${WD}/PhaseTwo_gdc
+    if [ $Phase2ndDir = TRUE ] ; then
+      ${FSLDIR}/bin/fslmaths ${WD}/PhaseOne2_gdc -mul ${WD}/PhaseOne2_gdc_warp_jacobian ${WD}/PhaseOne2_gdc
+      ${FSLDIR}/bin/fslmaths ${WD}/PhaseTwo2_gdc -mul ${WD}/PhaseTwo2_gdc_warp_jacobian ${WD}/PhaseTwo2_gdc
+    fi
   fi
   #overwrites inputs, no else needed
   
   #in the below stuff, the jacobians for both phases and sbref are applied unconditionally to a separate _jac image
   #NOTE: "SBref" is actually the input scout, which is actually the _gdc scout, with gdc jacobian applied if applicable
 
+  if [ $Phase2ndDir = TRUE ] ; then
+      Phase2ndSet="PhaseOne2 PhaseTwo2 PhaseOne2_gdc PhaseTwo2_gdc PhaseOne2_gdc_warp PhaseTwo2_gdc_warp"
+  else
+      Phase2ndSet=""
+  fi
+
+  if [[ ("$TruePatientPosition" = "HFSx" || "$TruePatientPosition" = "FFSx" || "$TruePatientPosition" = "HFS" || "$TruePatientPosition" = "FFS" ) && ( "$TruePatientPosition" != "$ScannerPatientPosition") ]] ; then
+    if [ ! -z "$InitWorldMat" ] ; then
+      log_Msg "Apply init rigid-body transformation to sform"
+      initmat="--init=${InitWorldMat}"
+    else
+      initmat=""
+    fi
+    log_Msg "Reorient $TruePatientPosition data with a scanner orientation of $ScannerPatientPosition"
+    for vol in PhaseOne PhaseTwo PhaseOne_gdc PhaseTwo_gdc PhaseOne_gdc_warp PhaseTwo_gdc_warp $Phase2ndSet; do
+  ${GlobalScripts}/CorrectVolumeOrientation --in=${WD}/${vol}.nii.gz --out=${WD}/${vol} --tposition=$TruePatientPosition --sposition=$ScannerPatientPosition $initmat
+    done
+
+  else
```

Comment:
This is something new for human, but it is probably okay as we have previously recommended people do this outside of the pipeline.